server/src/main/java/org/opensearch/index/codec/KnnVectorFormatFactory.java

```diff
@@ -0,0 +1,66 @@
+/*
+ *  Copyright OpenSearch Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.index.codec;
+
+import org.apache.lucene.codecs.KnnVectorsFormat;
+import org.apache.lucene.codecs.lucene92.Lucene92Codec;
+import org.apache.lucene.codecs.lucene92.Lucene92HnswVectorsFormat;
+import org.opensearch.index.mapper.DenseVectorFieldMapper;
+import org.opensearch.index.mapper.KnnAlgorithmContext;
+import org.opensearch.index.mapper.MappedFieldType;
+import org.opensearch.index.mapper.MapperService;
+
+import java.util.Map;
+
+import static org.opensearch.index.mapper.DenseVectorFieldMapper.DenseVectorFieldType;
+import static org.opensearch.index.mapper.KnnAlgorithmContextFactory.HNSW_PARAMETER_BEAM_WIDTH;
+import static org.opensearch.index.mapper.KnnAlgorithmContextFactory.HNSW_PARAMETER_MAX_CONNECTIONS;
+
+/**
+ * Factory that creates a {@link KnnVectorsFormat knn vector format} based on a mapping
+ * configuration for the field.
+ *
+ * @opensearch.internal
+ */
+public class KnnVectorFormatFactory {
```

Comment:
WDYT?
```suggestion
public class KNNVectorFormatFactory {
```

Reply:
I'm not sure to be honest - we're wrapping Lucene  classes and they have notation as "Knn" (e.g. https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/document/KnnVectorField.java), so I've follow them